src/common/fsst.cpp

```diff
@@ -1,30 +1,11 @@
-#include "duckdb/storage/string_uncompressed.hpp"
-#include "duckdb/common/types/value.hpp"
 #include "duckdb/common/fsst.hpp"
-#include "fsst.h"
 
 namespace duckdb {
 
-string_t FSSTPrimitives::DecompressValue(void *duckdb_fsst_decoder, Vector &result, const char *compressed_string,
-                                         const idx_t compressed_string_len, vector<unsigned char> &decompress_buffer) {
-
-	D_ASSERT(result.GetVectorType() == VectorType::FLAT_VECTOR);
-	auto fsst_decoder = reinterpret_cast<duckdb_fsst_decoder_t *>(duckdb_fsst_decoder);
-	auto compressed_string_ptr = (unsigned char *)compressed_string; // NOLINT
-	auto decompressed_string_size = duckdb_fsst_decompress(fsst_decoder, compressed_string_len, compressed_string_ptr,
-	                                                       decompress_buffer.size(), decompress_buffer.data());
-
-	D_ASSERT(!decompress_buffer.empty());
-	D_ASSERT(decompressed_string_size <= decompress_buffer.size() - 1);
-	return StringVector::AddStringOrBlob(result, const_char_ptr_cast(decompress_buffer.data()),
-	                                     decompressed_string_size);
-}
-
 string FSSTPrimitives::DecompressValue(void *duckdb_fsst_decoder, const char *compressed_string,
                                        const idx_t compressed_string_len, vector<unsigned char> &decompress_buffer) {
-
-	auto compressed_string_ptr = (unsigned char *)compressed_string; // NOLINT
-	auto fsst_decoder = reinterpret_cast<duckdb_fsst_decoder_t *>(duckdb_fsst_decoder);
+	auto compressed_string_ptr = reinterpret_cast<const unsigned char *>(compressed_string);
+	auto fsst_decoder = static_cast<duckdb_fsst_decoder_t *>(duckdb_fsst_decoder);
```

Comment:
While we're at it, perhaps we can replace the use of `void *` by forward-declaring `duckdb_fsst_decoder`

Reply:
Ah so it's already a `void*` underneath, but then I don't get why we are moving it around as a `void*` and then casting it to `duckdb_fsst_encoder_t *` in here, that's what I meant.

We should be able to use `duckdb_fsst_encoder_t *` everywhere instead of `void*`
We still need to forward declare though because we don't want to include third_party headers in core headers

Reply:
> This was done on purpose by Sam >2 years ago. From `fsst.h`:
> 
> ```c++
> /* Data structure needed for compressing strings - use duckdb_fsst_duplicate() to create thread-local copies. Use duckdb_fsst_destroy() to free. */
> typedef void* duckdb_fsst_encoder_t; /* opaque type - it wraps around a rather large (~900KB) C++ object */
> ```
> 
> It was purposely forward-declared as an opaque type there, so I would rather not touch this.

Actually, that's `encoder`, not `decoder`